pkg/test/e2e/ccm-migration/ccm_migration_test.go

```diff
@@ -40,117 +40,156 @@ import (
+			userClient = setupAndGetUserClient(clusterJig, openstackCluster, clusterClientProvider)
+		})
 
-			ginkgo.By("waiting for the complete cluster migration")
-			gomega.Expect(wait.Poll(userClusterPollInterval, customTestTimeout, func() (done bool, err error) {
-				migratingCluster := &kubermaticv1.Cluster{}
-				if err := clusterJig.SeedClient.Get(context.Background(), types.NamespacedName{Name: clusterJig.Cluster.Name}, migratingCluster); err != nil {
-					return false, err
-				}
+		ginkgo.AfterEach(func() {
+			gomega.Expect(clusterJig.Cleanup(userClient)).NotTo(gomega.HaveOccurred())
+		})
 
-				return migratingCluster.Status.Conditions[kubermaticv1.ClusterConditionCSIKubeletMigrationCompleted].Status == corev1.ConditionTrue, nil
-			})).NotTo(gomega.HaveOccurred())
+		ginkgo.It("migrating cluster to external CCM", func() {
+			testBody(clusterJig, openstackCluster, userClient)
 		})
 	})
 })
+
+func setupAndGetUserClient(clusterJig providers.ClusterJigInterface, cluster *kubermaticv1.Cluster, clusterClientProvider *clusterclient.Provider) ctrlruntimeclient.Client {
+	ginkgo.By("setting up cluster")
+	gomega.Expect(clusterJig.Setup()).NotTo(gomega.HaveOccurred(), "user cluster should deploy successfully")
+	clusterJig.Log().Debugw("Cluster set up", "name", clusterJig.Name())
+	gomega.Expect(clusterJig.Seed().Get(context.TODO(), types.NamespacedName{Name: clusterJig.Name()}, cluster)).NotTo(gomega.HaveOccurred())
+
+	var userClient ctrlruntimeclient.Client
+	gomega.Expect(wait.Poll(utils.UserClusterPollInterval, utils.CustomTestTimeout, func() (done bool, err error) {
+		gomega.Expect(clusterJig.Seed().Get(context.TODO(), types.NamespacedName{Name: clusterJig.Name()}, cluster)).NotTo(gomega.HaveOccurred())
+		userClient, err = clusterClientProvider.GetClient(context.TODO(), cluster)
+		if err != nil {
+			clusterJig.Log().Debug("user cluster client get failed: %v", err)
+			return false, nil
+		}
+		return true, nil
+	})).NotTo(gomega.HaveOccurred())
+	clusterJig.Log().Debugw("User cluster client correctly created")
+
+	gomega.Expect(clusterv1alpha1.AddToScheme(userClient.Scheme())).NotTo(gomega.HaveOccurred())
+
+	gomega.Expect(wait.Poll(utils.UserClusterPollInterval, utils.CustomTestTimeout, func() (done bool, err error) {
+		if err = clusterJig.CreateMachineDeployment(userClient); err != nil {
+			clusterJig.Log().Debug("machine deployment creation failed: %v", err)
+			return false, nil
+		}
+		return true, nil
+	})).NotTo(gomega.HaveOccurred())
+	clusterJig.Log().Debug("MachineDeployment created")
+
+	return userClient
+}
+
+func testBody(clusterJig providers.ClusterJigInterface, cluster *kubermaticv1.Cluster, userClient ctrlruntimeclient.Client) {
+	ginkgo.By("enabling externalCloudProvider feature")
+	gomega.Expect(clusterJig.Seed().Get(context.TODO(), types.NamespacedName{Name: clusterJig.Name()}, cluster)).NotTo(gomega.HaveOccurred())
```

Comment:
I feel like we repeat `clusterJig.Seed().Get`. Is it possible to somehow extract that in a function?

Reply:
Why have a wrapper to `Seed().Get()`? I don't understand what advantage it could introduce.

Reply:
The idea was to reduce duplication, but we can leave it as it is.